pkg/manifests/types.go

```diff
@@ -19,377 +19,398 @@ import (
 	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
-	// VolumeClaimTemplate defines persistent storage for Thanos Ruler. It's possible to
-	// configure storageClass and size of volume.
+	// Defines persistent storage for Thanos Ruler. Use this setting to
+	// configure the storage class and size of a volume.
 	VolumeClaimTemplate *monv1.EmbeddedPersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
 }
 
 // ----- Common Types -----
 
-// AdditionalAlertmanagerConfig defines configuration on how a component should communicate with
-// aditional Alertmanager instances.
+// `AdditionalAlertmanagerConfig` defines settings for how a component 
+// communicates with additional Alertmanager instances.
 type AdditionalAlertmanagerConfig struct {
-	// APIVersion defines the api version of Alertmanager.
+	// Defines the API version of Alertmanager.
 	APIVersion string `json:"apiVersion"`
-	// BearerToken defines the bearer token to use when authenticating to Alertmanager.
+	// Defines the bearer token to use when authenticating to Alertmanager.
 	BearerToken *v1.SecretKeySelector `json:"bearerToken,omitempty"`
-	// PathPrefix defines the path prefix to add in front of the push endpoint path.
+	// Defines the path prefix to add in front of the push endpoint path.
 	PathPrefix string `json:"pathPrefix,omitempty"`
-	// Scheme the URL scheme to use when talking to Alertmanagers.
+	// Defines the URL scheme to use when communicating with Alertmanager instances.
 	Scheme string `json:"scheme,omitempty"`
-	// StaticConfigs a list of statically configured Alertmanagers.
+	// A list of statically configured Alertmanager instances.
 	StaticConfigs []string `json:"staticConfigs,omitempty"`
-	// Timeout defines the timeout used when sending alerts.
+	// Defines the timeout value used when sending alerts.
 	Timeout *string `json:"timeout,omitempty"`
-	// TLSConfig defines the TLS Config to use for alertmanager connection.
+	// Defines the TLS settings to use for Alertmanager connections.
 	TLSConfig TLSConfig `json:"tlsConfig,omitempty"`
 }
 
-// RemoteWriteSpec is almost a 1to1 copy of monv1.RemoteWriteSpec but with the
-// BearerToken field removed. In the future other fields might be added here.
+// `RemoteWriteSpec` defines the settings for remote write storage.
 type RemoteWriteSpec struct {
-	// Authorization defines the authorization section for remote write
+	// Defines the authorization settings for remote write storage.
 	Authorization *monv1.SafeAuthorization `json:"authorization,omitempty"`
-	// BasicAuth defines configuration for basic authentication for the URL.
+	// Defines basic authentication settings for the remote write endpoint URL.
 	BasicAuth *monv1.BasicAuth `json:"basicAuth,omitempty"`
-	// BearerTokenFile defines the file where the bearer token for remote write resides.
+	// Defines the file that contains the bearer token for the remote write endpoint.
```

Comment:
This one we could skip from the documentation eventually because it's not particularly useful: users don't have ways to mount secrets in the pod so in practice they could only reference the service account's token. I'm not sure that the generation tool allows to "skip" a field though so this might require a follow-up.

Reply:
I changed this to `// Defines the file that contains the bearer token for the remote write 
	// endpoint.
	// However, because you cannot mount secrets in a pod, in practice
	// you can only reference the token of the service account.`